usl_models/usl_models/atmo_ml/datasets.py

```diff
@@ -1,192 +1,199 @@
-        spatial_file_name: GCS path for spatial data.
-        lu_index_file_name: GCS path for LU index.
-        batch_size: Batch size for prediction.
-        time_steps_per_day: Number of time steps per day for spatiotemporal data.
-        storage_client: The GCS client.
-
-    Yields:
-        A batch of prediction inputs.
-    """
-    bucket = storage_client.bucket(bucket_name)
-
-    # Load spatial and LU index data
-    spatial_blob = bucket.blob(spatial_file_name)
-    lu_index_blob = bucket.blob(lu_index_file_name)
-
-    spatial_data = np.load(spatial_blob.open("rb"))
-    lu_index_data = np.load(lu_index_blob.open("rb"))
-
-    # Load spatiotemporal data in batches for prediction
-    for st_file_name in spatiotemporal_file_names:
-        spatiotemporal_blob = bucket.blob(st_file_name)
-        spatiotemporal_data = np.load(spatiotemporal_blob.open("rb"))
-
-        # Split the spatiotemporal data into batches
-        for i in range(0, spatiotemporal_data.shape[0], batch_size):
-            batch_spatiotemporal = spatiotemporal_data[i : i + batch_size]
-
-            # Yield the batch of inputs
-            inputs = {
-                "spatiotemporal": batch_spatiotemporal,
-                "spatial": spatial_data,
-                "lu_index": lu_index_data.flatten(),
-            }
-            yield inputs
-
-
 def create_atmo_dataset(
-    bucket_name: str,
-    spatiotemporal_file_names: list[str],
-    label_file_names: list[str],
-    spatial_file_name: str,
-    lu_index_file_name: str,
+    data_bucket_name: str,
+    label_bucket_name: str,
+    spatiotemporal_folder: str,
+    spatial_folder: str,
+    lu_index_folder: str,
+    label_folder: str,
     time_steps_per_day: int,
-    batch_size: int = 32,
+    batch_size: int = 4,
     shuffle: bool = True,
     storage_client: storage.Client = None,
     firestore_client: firestore.Client = None,
-) -> tf.data.Dataset:
-    """Creates the dataset for the AtmoML model with optional Firestore logging.
-
-    Args:
-        bucket_name: The GCS bucket name.
-        spatiotemporal_file_names: List of file names for spatiotemporal in GCS.
-        label_file_names: List of file names for corresponding labels in the GCS.
-        spatial_file_name: File name for spatial features (static data) in the GCS.
-        lu_index_file_name: File name for the land use index data in the GCS.
-        time_steps_per_day: Number of time steps to divide per day.
-        batch_size: Batch size for the dataset.
-        shuffle: Whether to shuffle the dataset.
-        storage_client: An instance of Google Cloud Storage client.
-        firestore_client: An optional Firestore client to log data loading status.
-
-    Returns:
-        A tf.data.Dataset object yielding input dictionaries compatible with AtmoML.
-    """
+) -> tuple[tf.data.Dataset, tf.data.Dataset, tf.data.Dataset]:
+
+    def pad_or_truncate_data(data, target_length, pad_value=0):

```

Comment:
Why is this function nested? If it doesn't need to capture local variables then let's move it to module level.

Reply:
done